dotnet-peer/Libp2pChat/Infrastructure/Libp2p/Libp2pChatService.cs

```diff
@@ -0,0 +1,342 @@
+            _logger.LogInformation("Starting pubsub router...");
+            await _router.StartAsync(_localPeer, _cancellationTokenSource.Token);
+            _logger.LogInformation("Pubsub router started successfully");
+            
+            // Start cleanup task
+            _ = Task.Run(RunPeerCleanupTask, _cancellationTokenSource.Token);
+            
+            _isStarted = true;
+            
+            _logger.LogInformation("Chat service started successfully");
+            _logger.LogInformation($"Peer ID: {PeerId}");
+            _logger.LogInformation($"GO PEER CONNECTION COMMAND:");
+            _logger.LogInformation($"./go-peer --connect /ip4/127.0.0.1/tcp/9096/p2p/{PeerId}");
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError("Failed to start chat service", ex);
+            await StopAsync();
+            throw;
+        }
+    }
+    
+    /// <inheritdoc />
+    public async Task StopAsync()
+    {
+        if (!_isStarted)
+            return;
+            
+        _logger.LogInformation("Stopping chat service...");
+        
+        try
+        {
+            if (_topic != null)
+            {
+                _topic.OnMessage -= HandleMessageReceived;
+            }
+            
+            if (_cancellationTokenSource != null)
+            {
+                if (!_cancellationTokenSource.IsCancellationRequested)
+                    _cancellationTokenSource.Cancel();
+                    
+                _cancellationTokenSource.Dispose();
+                _cancellationTokenSource = null;
+            }
+            
+            _isStarted = false;
+            _logger.LogInformation("Chat service stopped");
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError("Error while stopping chat service", ex);
+            throw;
+        }
+    }
+    
+    /// <inheritdoc />
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+    
+    /// <summary>
+    /// Disposes resources.
+    /// </summary>
+    /// <param name="disposing">Whether to dispose managed resources.</param>
+    protected virtual void Dispose(bool disposing)
+    {
+        if (_isDisposed)
+            return;
+            
+        if (disposing)
+        {
+            _cancellationTokenSource?.Dispose();
+        }
+        
+        _isDisposed = true;
+    }
+    
+    private void HandleMessageReceived(byte[] data)
+    {
+        try
+        {
+            if (data == null || data.Length == 0)
+            {
+                _logger.LogWarning("Received empty message data");
+                return;
+            }
+            
+            string raw = Encoding.UTF8.GetString(data).Trim();
+            _logger.LogDebug($"Raw message received: {raw}");
+            
+            // Check if the message is JSON or plain text
+            bool isJsonMessage = raw.StartsWith("{");
+            if (isJsonMessage)
```

Comment:
Not a good check, you can omit it and catch potential error in existing catch block below